core/src/main/java/org/elasticsearch/index/translog/Checkpoint.java

```diff
@@ -33,25 +34,40 @@
 
 /**
  */
-class Checkpoint {
+public class Checkpoint {
+
+    static final byte FORMAT_VERSION = 1;
 
    static final int BUFFER_SIZE = Integer.BYTES  // ops
             + Long.BYTES // offset
-            + Long.BYTES;// generation
+            + Long.BYTES // generation
+            + Long.BYTES; // seq no global checkpoint
     final long offset;
     final int numOps;
     final long generation;
+    final long seqNoGlobalCheckpoint;
 
-    Checkpoint(long offset, int numOps, long generation) {
+    Checkpoint(long offset, int numOps, long generation, long seqNoGlobalCheckpoint) {
```

Comment:
ok so this is just wrong, the `seqNoGlobalCheckpoint` has nothing to do with the checkpoint here. We can't allow arbitrary metadata to be written into this file. This is an 100% no-go. Can we model this differently by maybe adding a special translog operation that gets added into the translog before we sync? so we can just open the translog based on the checkpoint and read the operation from the outside? I don't want to mix concerns here in the translog.